pghoard/pghoard.py

```diff
@@ -609,6 +621,33 @@ def startup_walk_for_missed_files(self):
 
                 filetype = FileType.Timeline if wal.TIMELINE_RE.match(filename) else FileType.Wal
 
+                # verify if file was already compressed, otherwise the transfer agent will encounter
+                # duplicated UploadEvents. In case it was compressed, we should just add it to the deletion queue
+                base_compressed_file_path = (
+                    compressed_timeline_path if filetype is FileType.Timeline else compressed_xlog_path
+                )
+                compressed_file_path = os.path.join(base_compressed_file_path, filename)
+                is_already_compressed = os.path.exists(compressed_file_path)
+                has_metadata_file = os.path.exists(compressed_file_path + ".metadata")
+
+                # the file was compressed correctly
+                if is_already_compressed and has_metadata_file:
+                    self.log.debug("Uncompressed file %r is already compressed, adding to deletion queue.", full_path)
+                    if filetype is FileType.Timeline:
+                        os.unlink(full_path)
+                    else:
+                        delete_request = WalFileDeletionEvent(backup_site_name=site, file_path=Path(full_path))
+                        self.wal_file_deletion_queue.put(delete_request)
+                        self.log.info("Adding to Uncompressed WAL file to deletion queue: %s", full_path)
+                    continue
+
+                # delete compressed file and re-try
+                if is_already_compressed and not has_metadata_file:
+                    self.log.info(
+                        "Deleting invalid compressed file %r, compression will be re-tried", compressed_file_path,
```

Comment:
nit: when one sees this message, most probably the guess would be some corruption or something. I'd just put "Deleting incomplete compressed file (missing metadata), compression will be re-tried"

Reply:
Fixed here [db82d1a](https://github.com/aiven/pghoard/pull/592/commits/db82d1a19fa8c2c79796942e52a4c36156375a35)